### PR TITLE
TAJO-923: Add VAR_SAMP and VAR_POP window functions

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
I coded in the same way in https://github.com/apache/tajo/pull/427 (TAJO-921)
If needed, after merging pull-request 427, I will do refactoring.

This code is passed the following command.
```
mvn clean install -Pparallel-test,hcatalog-0.12.0 -DLOG_LEVEL=ERROR -Dmaven.fork.count=2 &> TAJO-923.travis.ERROR.txt
```
The result file is https://app.box.com/s/clnx9d2bisfhbh2upgo3q6g7bdx7aic5

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
Rebased.

**Comment by dongjoon-hyun** [comment]
Rebased.

**Comment by jihoonson** [comment]
I'll review today.

**Comment by jihoonson** [comment]
The only difference between this patch and TAJO-921 is the absence of square root. Why should we maintan mostly duplicated codes separately?

**Comment by jihoonson** [comment]
No offense. Sorry for late review.

**Comment by dongjoon-hyun** [comment]
@jihoonson , I knew that 20 days ago. :) But, there was no way to implement variance at that time, since TAJO-921 is not committed yet. As I mentioned my first comment, we need to change. 

Here, I need your guide how to reduce the redundancy. Is it proper to compute STD_DEV first and square that?

**Comment by jihoonson** [comment]
Thanks for your rapid reply. 
Yes, I also think that will be a good way.

**Comment by dongjoon-hyun** [comment]
Hi, @jihoonson . I removed the following redundancy. Now, VarSamp and VarPop use StdDevContext.
* ProtocolBuffer message for VarSamp/VarPop
* Var.java file

File: tajo-core/src/main/java/org/apache/tajo/engine/function/builtin/VarPop.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+package org.apache.tajo.engine.function.builtin;
+
+import org.apache.tajo.catalog.Column;
+import org.apache.tajo.datum.Datum;
+import org.apache.tajo.datum.DatumFactory;
+import org.apache.tajo.datum.NullDatum;
+import org.apache.tajo.plan.function.FunctionContext;
+
+public abstract class VarPop extends StdDev {
```
Comment (jihoonson) [inline tajo-core/src/main/java/org/apache/tajo/engine/function/builtin/VarPop.java]: I think that we need to change ```StdDev``` to more general name.
How about ```Variance```? Or, do you have any good idea?

Comment (dongjoon-hyun) [inline tajo-core/src/main/java/org/apache/tajo/engine/function/builtin/VarPop.java]: It sounds good to me, too. I agree that `Variance` looks more general to cover all VarPop/VarSamp/StdDevPop/StdDevSamp functions. Then, `StdDevContext` should be changed into `VarianceContext` accordingly.

Comment (dongjoon-hyun) [inline tajo-core/src/main/java/org/apache/tajo/engine/function/builtin/VarPop.java]: @jihoonson , I changed the code according to your comments. Thank you always.

**Comment by jihoonson** [comment]
+1 thanks for your work.

**Pull request closed without merging**